dags/process_elife_zip_dag.py

```diff
@@ -0,0 +1,107 @@
+"""
+DAG processes a zip file stored in an AWS S3 bucket and prepares the extracted xml
+file for Libero content API and sends to the content store via a PUT request.
+"""
+import logging
+from datetime import timedelta
+from io import BytesIO
+from tempfile import TemporaryFile
+from zipfile import ZipFile
+
+from airflow import DAG, configuration
+from airflow.operators import python_operator
+from airflow.utils import timezone
+from lxml import etree
+
+from aws import get_aws_connection, list_bucket_keys_iter
+
+SOURCE_BUCKET = configuration.conf.get('libero', 'source_bucket')
+DESTINATION_BUCKET = configuration.conf.get('libero', 'destination_bucket')
+
+logger = logging.getLogger(__name__)
+
+default_args = {
+    'owner': 'libero',
+    'depends_on_past': False,
+    'start_date': timezone.utcnow(),
+    'email': ['libero@example.com'],
+    'email_on_failure': False,
+    'email_on_retry': False,
+    'retries': 3,
+    'retry_delay': timedelta(seconds=5)
+}
+
+
+def extract_zipped_files_to_bucket(**context):
+    # get file name passed from trigger
+    dag_run = context['dag_run']
+    conf = dag_run.conf or {}
+    file_name = conf.get('file')
+    logger.debug('FILE PASSED= %s', file_name)
+    if not file_name:
+        raise ValueError('%s triggered without a file name passed to conf' % dag_run.dag_id)
+
+    # temporary files are securely stored on disk and automatically deleted when closed
+    with TemporaryFile() as temp_file:
+        s3 = get_aws_connection('s3')
+        # store downloaded file in temp file
+        s3.download_fileobj(SOURCE_BUCKET, file_name, temp_file)
+        logger.debug('ZIPPED FILES= %s', ZipFile(temp_file).namelist())
+
+        folder_name = file_name.replace('.zip', '')
+        xml_path = None
+        for zipped_file_path in ZipFile(temp_file).namelist():
+            # skip mac os archive files
+            if '__MACOSX' not in zipped_file_path:
+                with TemporaryFile() as inner_temp_file:
+                    # store zipped file in temp file
+                    inner_temp_file.write(ZipFile(temp_file).read(zipped_file_path))
+                    inner_temp_file.seek(0)
+                    s3_key = '%s/%s' % (folder_name, zipped_file_path)
+                    # upload stored zipped file preserving zip file structure
+                    s3.upload_fileobj(inner_temp_file, DESTINATION_BUCKET, s3_key)
+                    logger.info(
+                        '%s uploaded to %s/%s', zipped_file_path, DESTINATION_BUCKET, s3_key
+                    )
+                    if zipped_file_path.endswith('.xml'):
```

Comment:
can there be more than 1 xml in the same zip? Not sure, but I would assert that there is only one so that an exception is thrown if there is a new case where >=2 are used

Reply:
Yes I also made an assumption here that there would only be one xml file. I'm unsure if there can be multiple xml files. Do you know if `manifest.xml` files are included sometimes?

Reply:
As far as I know there are no manifests at this stage (I've seen them in MECA, a format used by xpub but not here.)

Reply:
Good to assume that there is only one file, would assert that this is the case.

Reply:
check added.